Source/SwiftLintFramework/Rules/MultilineFunctionChainsRule.swift

```diff
@@ -0,0 +1,221 @@
+            let chain = a.b(1, 2, 3)
+                .c(.init(
+                    a: 1,
+                    b, 2,
+                    c, 3))
+                .d()
+            """,
+            """
+            self.viewModel.outputs.postContextualNotification
+              .observeForUI()
+              .observeValues {
+                NotificationCenter.default.post(
+                  Notification(
+                    name: .ksr_showNotificationsDialog,
+                    userInfo: [UserInfoKeys.context: PushNotificationDialog.Context.pledge,
+                               UserInfoKeys.viewController: self]
+                 )
+                )
+              }
+            """,
+            "let remainingIDs = Array(Set(self.currentIDs).subtracting(Set(response.ids)))",
+            """
+            self.happeningNewsletterOn = self.updateCurrentUser
+                .map { $0.newsletters.happening }.skipNil().skipRepeats()
+            """
+        ],
+        triggeringExamples: [
+            """
+            let evenSquaresSum = [20, 17, 35, 4]
+                .filter { $0 % 2 == 0 }↓.map { $0 * $0 }
+                .reduce(0, +)
+            """,
+            """
+            let evenSquaresSum = a.b(1, 2, 3)
+                .c { blah in
+                    print(blah)
+                }↓.d()
+            """,
+            """
+            let evenSquaresSum = a.b(1, 2, 3)
+                .c(2, 3, 4)↓.d()
+            """,
+            """
+            let evenSquaresSum = a.b(1, 2, 3)↓.c { blah in
+                    print(blah)
+                }
+                .d()
+            """
+        ]
+    )
+
+    public func validate(file: File,
+                         kind: SwiftExpressionKind,
+                         dictionary: [String: SourceKitRepresentable]) -> [StyleViolation] {
+        return violatingOffsets(file: file, kind: kind, dictionary: dictionary).map { offset in
+            return StyleViolation(ruleDescription: type(of: self).description,
+                                  severity: configuration.severity,
+                                  location: Location(file: file, characterOffset: offset))
+        }
+    }
+
+    private func violatingOffsets(file: File,
+                                  kind: SwiftExpressionKind,
+                                  dictionary: [String: SourceKitRepresentable]) -> [Int] {
+        let ranges = callRanges(file: file, kind: kind, dictionary: dictionary)
+
+        let calls = ranges.compactMap { range -> (dotLine: Int, dotOffset: Int, range: NSRange)? in
+            guard
+                let offset = callDotOffset(file: file, callRange: range),
+                let line = file.contents.bridge().lineAndCharacter(forCharacterOffset: offset)?.line else {
+                    return nil
+            }
+            return (dotLine: line, dotOffset: offset, range: range)
+        }
+
+        let uniqueLines = calls
+            .map { $0.dotLine }
```

Comment:
we have an `unique` computed property declared on an `Array` extension, probably we could use it here